### ISSUE TITLE
Feature Request: Hydrate some events with additional data

### ISSUE THREAD

**Issue by BarryCarlyon** [obsproject/obs-websocket#983]
### Feature Request Type

RPC Event

### Requested Feature

Comparing v4 to v5 a number of events are "dry" on the data they send.

For example `SceneTransitionStarted` only decalres the kind of transition that is occuring.

The v4 equivalent being `TransitionBegin` decarles additional information with the event including the `duration` and `from`/`to` scenes

I imaging there are similar "problems" with other events being "dry" on data when compared to their v4 counter parts. But I'm just getting started on planning my OBS 27 to 28 migration paths.

I will likely update this issue with other "dry" events as I poke around

### Requested Feature Usage Scenario

In the example of `SceneTransitionStarted`.

I detect the duration of the event and the destination scene.
Half way thru the transition I run actions or tasks based on the destination scene.

So I need this information _as the transitions starts_

**Comment by deeoncomms** [comment]
I agree with this - I set up some events for an obs scene collection to show or hide certain elements of a nested scene depending on the live scene, and because of that setup, anything other than a cut transition looks awkward using the SwitchScenes (4.x) or CurrentProgramSceneChanged (5.x) events due to the source visibility change happening after the transition is over.

**Comment by ChristopheCVB** [comment]
Same for `SceneTransitionEnded`, lacks the `toScene`

**Comment by tinodo** [comment]
Okay, another +1 for this. Getting the `fromScene `and `toScene `is rather important to me as well. I used this very frequently and stops me from upgrading to the latest version.

**Comment by cai-tan** [comment]
Actually kind of insane this still hasn't been updated an entire year later. Why was `fromScene` and `toScene` even removed in the first place?

**Comment by zoton2** [comment]
Should this issue be broken down into more issues? Did @BarryCarlyon find any more issues in migration? I'm currently migrating and would really like to have at least the to/from scenes in `SceneTransitionStarted`. 4.x did that with a custom function, [GetTransitionData](https://github.com/obsproject/obs-websocket/blob/4.x-compat/src/Utils.cpp#L479), but I think the data/APIs it uses do still exist in OBS, just not sure how hacky this custom function was.

**Comment by duncte123** [comment]
Created a PR for this functionality, still looking into setting the duration on the response but the current patch works like a charm https://github.com/obsproject/obs-websocket/pull/1229

**Comment by duncte123** [comment]
Little update from my side, here are windows and linux binaries if you want to use this already. These are the binary files only since the language files have not updated (you can always compile the PR yourself if you don't trust these downloads)

[obs-websocket-linux.zip](https://github.com/user-attachments/files/15710715/obs-websocket-linux.zip)
[obs-websocket-windows-x64.zip](https://github.com/user-attachments/files/15710720/obs-websocket-windows-x64.zip)

**Comment by duncte123** [comment]
I've created a new build of the websocket DLL that is based on 5.5.3
[obs-websocket-win-x64.zip](https://github.com/user-attachments/files/17111640/obs-websocket-win-x64.zip)
